auth/src/api/handlers.rs

```diff
@@ -51,12 +52,29 @@ pub(crate) async fn login(
         .expect("to set account name");
     session
         .insert("account_tier", user.account_tier)
-        .expect("to set account name");
+        .expect("to set account tier");
 
     Ok(Json(user.into()))
 }
 
-pub(crate) async fn logout(mut session: WritableSession) {
+pub(crate) async fn logout(
+    mut session: WritableSession,
+    State(cache_manager): State<CacheManagerState>,
+    headers: HeaderMap,
+) {
+    // If there is a cookie, extract it and try to get the id.
+    let cache_key = if let Ok(Some(cookie)) = headers.typed_try_get::<Cookie>() {
+        cookie.get("shuttle.sid").map(|id| id.to_string())
+    } else {
+        None
+    };
+
+    // If there was an id in the cookie, clear it from the cache.
+    if let Some(key) = cache_key {
+        // Clear the session's associated JWT from the cache.
+        cache_manager.invalidate(&key);
+    }
```

Comment:
We should be able tto combine the two ifs here - ie removing the `cache_key` variable.

Reply:
Ah, indeed! This part of logout was removed in the move to gateway though, we'll need to invalidate it there instead, either in its own layer or in the auth layer.